src/abi.h

```diff
@@ -45,6 +45,12 @@ typedef struct dhcp_overrides {
     guint metric;
 } NetplanDHCPOverrides;
 
+typedef struct ipv6_ra_overrides {
+    gboolean use_dns;
+    char* use_domains;
```

Comment:
`use_domains` should probably be an `enum`. I know that it's also a string in `dhcp_overrides` but I feel like we should fix it too (that can be done later). I can't think of a reason why it's a `char*` as there is only 3 values accepted by these options (true, false or route).

We'd need a new `handle_netdef_something` to handle it and it could also be used later in the dhcp overrides case.

Besides that, we need to introduce some validation for this option (including the dhcp4_overrides that was already there). The configuration below will be accepted by the parser for example.

```
network:
  version: 2
  ethernets:
    eth0:
      dhcp4-overrides:
        use-domains: thisisnotavalidoption
      ipv6-ra-overrides:
        use-domains: thisisnotavalidoption
```

In this case the parser should error out with a helpful error message saying it only accepts true, false or route.